bin/git-jira-branch.rb

```diff
@@ -0,0 +1,197 @@
+
+    if transition
+      # Seems we need to do this in two updates.  JIRA complained if we tried to transition and update fields.
+      attrs = {
+        update: {
+          comment: [{
+                      add: {
+                        body: "Created branch #{branch_name}"
+                      }
+                    }],
+          assignee: [{
+                       set: {
+                         name: @username
+                       }
+                     }]
+        }
+      }
+      issue.save!(attrs)
+      attrs = {
+        transition: {
+          id: transition
+        }
+      }
+      issue.client.post("#{issue.url}/transitions", attrs.to_json)
+    else
+      attrs = {
+        update: {
+          assignee: [{
+                       set: {
+                         name: @username
+                       }
+                     }],
+          comment: [{
+                      add: {
+                        body: "Created branch #{branch_name}"
+                      }
+                    }]
+        }
+      }
+      issue.save!(attrs)
+    end
+
+    puts Term::ANSIColor.green('Created branch and started JIRA issue')
+  end
+
+  private
+
+  def usage
+    'Usage: git jira-branch JIRA_ID'
+  end
+
+  def show_and_exit(message)
+    puts message
+    exit 1
+  end
+
+  def option_parser
+    @option_parser ||= OptionParser.new do |op|
+      op.banner = usage
+
+      op.on_tail('-h', '--help', 'Show this message') do
+        show_and_exit op
+      end
+    end
+  end
+
+  def get_jira_info(story_id)
+    @username = `git config jira.username`.strip
+    if @username.empty?
+      puts Term::ANSIColor.yellow %Q{
+        Your branch appears to have a story ID,
+        but I don't know your JIRA username!
+        Please set it with:
+        $ git config [--global] jira.username <token>
+      }
+      die "Aborting."
+    end
+
+    password = `git config jira.password`.strip
+    if password.empty?
+      puts Term::ANSIColor.yellow %Q{
+        Your branch appears to have a story ID,
+        but I don't know your JIRA password!
+        Please set it with:
+        $ git config [--global] jira.password <password>
+      }
+      die "Aborting."
+    end
+
+    site = `git config jira.site`.strip
+    if site.empty?
+      puts Term::ANSIColor.yellow %Q{
+        Your branch appears to have a story ID,
+        but I don't know your JIRA site!
+        Please set it with:
+        $ git config [--global] jira.site <https://mydomain.atlassian.net>
+      }
+      die "Aborting."
+    end
+    log.info "Finding your project and story¬"
+
+    options = {
+      username: @username,
+      password: password,
+      site: site,
+      context_path: '',
+      auth_type: :basic,
+      read_timeout: 120
+    }
+
+    client = JIRA::Client.new(options)
+    issue = client.Issue.find(story_id)
+
+    log.info '.'
+
+    log.info "Found story #{story_id} in '#{issue.fields['project']['name']}'"
```

Comment:
I renamed them all to `issue` and `issue_id`

On Wed, Jan 20, 2016 at 4:02 PM, Pedro Cunha <notifications@github.com>
wrote:

> In bin/git-jira-branch.rb
> <https://github.com/mezis/git-whistles/pull/47#discussion_r50326688>:
>
> > +
> > +    options = {
> > +      username: @username,
> > +      password: password,
> > +      site: site,
> > +      context_path: '',
> > +      auth_type: :basic,
> > +      read_timeout: 120
> > +    }
> > +
> > +    client = JIRA::Client.new(options)
> > +    issue = client.Issue.find(story_id)
> > +
> > +    log.info '.'
> > +
> > +    log.info "Found story #{story_id} in '#{issue.fields['project']['name']}'"
>
> I think if they are issues we should probably rename all "story"
> occurrences in this file to "issue" ? Or do you also use the word "story"
> to refer to JIRA issues ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mezis/git-whistles/pull/47/files#r50326688>.
>